backend/api/v1/schema_design_transformer.go

```diff
@@ -1422,7 +1435,859 @@ func checkDatabaseMetadata(engine v1pb.Engine, metadata *v1pb.DatabaseMetadata)
 	return nil
 }
 
+func checkColumnType(engine v1pb.Engine, tp string) bool {
+	switch engine {
+	case v1pb.Engine_MYSQL:
+		return checkMySQLColumnType(tp)
+	case v1pb.Engine_TIDB:
+		return checkTiDBColumnType(tp)
+	default:
+		return false
+	}
+}
+
 func checkMySQLColumnType(tp string) bool {
 	_, err := parser.ParseMySQL(fmt.Sprintf("CREATE TABLE t (a %s NOT NULL)", tp))
 	return err == nil
 }
+
+func checkTiDBColumnType(tp string) bool {
```

Comment:
nit: consider moving tidb code to a different go file. Do the same for mysql.

Reply:
Do you mean all the code about tidb? I do this because some types are not exported, such as `tableState`

Reply:
No, just the change in v1 package should stay in a standalone Go file within the same package. The file is too large.